ctera/src/main/java/ch/cyberduck/core/ctera/CteraAclPermissionFeature.java

```diff
@@ -0,0 +1,197 @@
+package ch.cyberduck.core.ctera;
+
+import ch.cyberduck.core.Acl;
+import ch.cyberduck.core.DefaultIOExceptionMappingService;
+import ch.cyberduck.core.DisabledListProgressListener;
+import ch.cyberduck.core.Local;
+import ch.cyberduck.core.LocaleFactory;
+import ch.cyberduck.core.Path;
+import ch.cyberduck.core.PathAttributes;
+import ch.cyberduck.core.dav.DAVExceptionMappingService;
+import ch.cyberduck.core.dav.DAVPathEncoder;
+import ch.cyberduck.core.dav.DAVSession;
+import ch.cyberduck.core.exception.AccessDeniedException;
+import ch.cyberduck.core.exception.BackgroundException;
+import ch.cyberduck.core.features.AclPermission;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.w3c.dom.Element;
+
+import javax.xml.namespace.QName;
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.github.sardine.impl.SardineException;
+import com.github.sardine.util.SardineUtil;
+
+
+public class CteraAclPermissionFeature implements AclPermission {
+    private static final Logger log = LogManager.getLogger(CteraAclPermissionFeature.class);
+
+    public static final String CTERA_NAMESPACE_URI = "http://www.ctera.com/ns";
+    public static final String CTERA_NAMESPACE_PREFIX = "ctera";
+
+    /**
+     * Read Data: Allows or denies viewing data in files.
+     */
+    public static final Acl.Role READPERMISSION = new Acl.Role("readpermission");
+
+    /**
+     * Write Data: Allows or denies making changes to a file and overwriting existing content.
+     */
+    public static final Acl.Role WRITEPERMISSION = new Acl.Role("writepermission");
+
+    /**
+     * Execute File: Allows or denies running program (executable) files.
+     * Files only.
+     */
+    public static final Acl.Role EXECUTEPERMISSION = new Acl.Role("executepermission");
+
+    /**
+     * Allows or denies deleting the file or folder. If you don't have Delete permission on a file or folder,
+     * you can still delete it if you have been granted Delete Subfolders and Files on the parent folder.
+     */
+    public static final Acl.Role DELETEPERMISSION = new Acl.Role("deletepermission");
+
+    /**
+     * Traverse Folder: Allows or denies moving through a restricted folder to reach files and folders
+     * beneath the restricted folder in the folder hierarchy. Traverse folder takes effect only when the group or user
+     * is not granted the "Bypass traverse checking user" right in the Group Policy snap-in.
+     * This permission does not automatically allow running program files.
+     * Directories only.
+     */
+    public static final Acl.Role TRAVERSEPERMISSION = new Acl.Role("traversepermission");
```

Comment:
Support for this role can be removed as it is not used.

Reply:
1a32ebe466b9b3b6caac80a8d0f8aeb272718965